### ISSUE TITLE
Add transactions and bank accounts

### ISSUE THREAD

**Issue by franz-josef-kaiser** [InvoiceShelf/InvoiceShelf#216]
**Is your feature request related to a problem? Please describe.**
The main problem is matching invoices to transactions from a bank account, which has to be done manually.

**Describe the solution you'd like**
In an optimal world, there would be a model/ entity for transactions. An easy and EU-compliant way to import data from bank accounts (for free) is using [GoCardless Bank Account Data](https://gocardless.com/bank-account-data/) service, which is what is integrated into InvoiceNinja (see [their docs](https://invoiceninja.github.io/en/transactions/)) as well.

Here's [a Google Doc](https://docs.google.com/spreadsheets/d/1EZ5n7QDGaRIot5M86dwqd5UFSGEDTeTRzEq3D9uEDkM/edit?gid=976380583#gid=976380583) that shows the bank integrations that they have. Another list [is available on their site](https://gocardless.com/bank-account-data/coverage/).

**Describe alternatives you've considered**
A CSV upload for bank account exports of transactions (as a first step into this direction), maybe.

**Comment by johnnyq** [comment]
Absolutely. This is exactly what’s needed, especially the ability to transfer funds between accounts.

This is necessary for proper accounting, particularly when a business has multiple account types, such as a Stripe account, bank account, and credit card account.

For example, invoices paid through Stripe should be recorded as income going into the Stripe account, while the Stripe processing fee should be recorded as an expense against the Stripe account.

The app should also allow you to transfer money from the Stripe account to the bank account, such as making a monthly transfer directly within the app.

Company expenses would be recorded against the credit card account. Then, when the credit card needs to be paid, you should be able to record a transfer from the bank account to the credit card account to reflect the payment.

Having the ability to transfer funds between accounts would make the accounting much more accurate and provide a complete picture of how money moves between Stripe, the bank account, and the credit card.

**Comment by johnnyq** [comment]
For good examples take a look at Akaunting, ITFlow, they do this very well. This would make this app extremely useful for accounting in general and much better book keeping system
